edit_course/exercise_forms.py

```diff
@@ -96,12 +96,25 @@ class RevealRuleForm(FieldsetModelForm):
 
     class Meta:
         model = RevealRule
-        fields = ['trigger', 'delay_minutes', 'time', 'currently_revealed']
+        fields = ['trigger', 'delay_minutes', 'time', 'currently_revealed', 'show_zero_points_immediately']
         widgets = {'time': DateTimeLocalInput}
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
+    def __init__(self, *args: Any, is_submission_feedback: bool = False, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
         self.fields['trigger'].widget.attrs['data-trigger'] = True
+
+        if is_submission_feedback:
+            self.fields['show_zero_points_immediately'].widget.attrs['data-visible-triggers'] = [
+                RevealRule.TRIGGER.DEADLINE.value,
+                RevealRule.TRIGGER.DEADLINE_ALL.value,
+                RevealRule.TRIGGER.DEADLINE_OR_FULL_POINTS.value,
+                RevealRule.TRIGGER.TIME.value,
+                RevealRule.TRIGGER.MANUAL.value,
+                RevealRule.TRIGGER.COMPLETION.value,
+            ]
+        else:
+            self.fields['show_zero_points_immediately'].widget.attrs['data-visible-triggers'] = [
+            ]
```

Comment:
I think this could be written differently to save code duplication:
```
triggers = [
    RevealRule.TRIGGER.DEADLINE.value,
    RevealRule.TRIGGER.DEADLINE_ALL.value,
    RevealRule.TRIGGER.DEADLINE_OR_FULL_POINTS.value,
    RevealRule.TRIGGER.TIME.value,
    RevealRule.TRIGGER.MANUAL.value,
    RevealRule.TRIGGER.COMPLETION.value,
] if is_submission_feedback else []
self.fields['show_zero_points_immediately'].widget.attrs['data-visible-triggers'] = triggers
```